include/onnxruntime/core/framework/op_kernel_shared.h

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "boost/mp11.hpp"
+
+namespace onnxruntime {
+
+std::unique_ptr<OpKernelInfo> CopyOpKernelInfo(const OpKernelInfo& info);
+
+class OpKernel {
+ public:
```

Comment:
Should this file be called op_kernel.h and the other one rename to op_kernel_context.h?